File: Functions/Update_Budget.c
#include <stdio.h>
float income, rent, utilities, groceries, transportation, savings, expenses, spend;

float input(char type[], float var){
    printf("monthly %s:\n", type);
    scanf("%f", &var);
    return var;
}


void percent(char type[], int amount){
    int per = amount/income *100;

    printf("your %s is %d%%" " of your income.\n", type, per);
}


int main(void){
    printf("This is a budget calculator.\n");
    income = input("income", income);
    rent = input("rent", rent);
    utilities = input("utilities", utilities);
    groceries = input("groceries", groceries);
    transportation = input("transportation", transportation);
    expenses = rent + utilities + groceries + transportation;
    savings = income * .2;
    spend = income - expenses - savings;
    printf("You make %.2f\n", income);
    printf("Your expenses are %.2f\n", expenses);
    printf("Your savings are %.2f\n", savings);
    printf("Your spending money is %.2f\n", spend);
    percent("rent", rent);
    percent("utilities", utilities);
    percent("groceries", groceries);
    percent("transportation", transportation);
    percent("expenses", expenses);
    percent("savings", savings);
    percent("spending money", spend);
  
    return 0;
}

File: Functions/update_budget.py
def input_value(type):
    var = float(input(f"monthly {type}:\n"))
    return var

def percent(type, amount, income):
    per = (amount / income) * 100
    print(f"your {type} is {per:.0f}% of your income.")

def main():
    print("This is a budget calculator.")
    income = input_value("income")
    rent = input_value("rent")
    utilities = input_value("utilities")
    groceries = input_value("groceries")
    transportation = input_value("transportation")
    
    expenses = rent + utilities + groceries + transportation
    savings = income * 0.2
    spend = income - expenses - savings
    
    print(f"You make {income:.2f}")
    print(f"Your expenses are {expenses:.2f}")
    print(f"Your savings are {savings:.2f}")
    print(f"Your spending money is {spend:.2f}")
    
    percent("rent", rent, income)
    percent("utilities", utilities, income)
    percent("groceries", groceries, income)
    percent("transportation", transportation, income)
    percent("expenses", expenses, income)
    percent("savings", savings, income)
    percent("spending money", spend, income)

if __name__ == "__main__":
    main()

